crates/matrix-sdk/src/attachment.rs

```diff
@@ -397,12 +394,14 @@ pub fn generate_image_thumbnail<R: BufRead + Seek>(
     thumbnail.write_to(&mut Cursor::new(&mut data), image_format)?;
     let data_size = data.len() as u32;
 
-    Ok((
-        data,
-        BaseThumbnailInfo {
-            width: Some(thumbnail_width.into()),
-            height: Some(thumbnail_height.into()),
-            size: Some(data_size.into()),
-        },
-    ))
+    let content_type = mime::Mime::from_str(image_format.to_mime_type())
+        .expect("image should give a valid mimetype");
```

Comment:
Can we avoid the `expect` and return an error instead?

Reply:
Changed